### ISSUE TITLE
fontbakery fixes: add ligature carets

### ISSUE THREAD

**Issue by arrowtype** [arrowtype/recursive#310]
Before we can merge into Google Fonts, it will be necessary to run the variable font through [FontBakery](https://github.com/googlefonts/fontbakery/) and note which checks will be important to follow. A few are semi-flexible "opinions" (e.g. UPM should equal 2000), while others are to solve recorded bugs (e.g. certain tables should be added to avoid issues in various software).

We will need to note down (and then make a script to fix) valid issues.

**Comment by benkiel** [comment]
Changes needed:

- [ ] VF font name change to Recursive[axislist].ttf
- [ ] Set fsType to 0
- [ ] Need license, copyright, etc info
- [ ] Set fsType to 0
- [ ] Add `gasp` table
- [ ] Fix default font name
- [ ] Add `prep` table
- [ ] Add ligature carets
- [ ] Shorten these glyph names: `quotesingle_quotesingle_quotesingle.code, hyphen_space_bracketleft_space_bracketright.code, hyphen_space_bracketleft_x_bracketright.code, numbersign_numbersign_numbersign.code, numbersign_numbersign_numbersign_numbersign.code and ampersand_ampersand_ampersand.code`
- [ ] Add `dsig`

Issues with font bakery that are going to come up:

* Currently barfing on the "MONO" axis name in the `varfont_instance_coordinates` check
* Instance naming; doesn't seem to be set up for the case of subfamilies (i.e. `"Mono Linear Light" is incorrect. It should be "Light"`)
* `MVAR` table: is this check still valid?

Issues to ignore(?)

* UPM at 2000 instead of 1000

**Comment by arrowtype** [comment]
Thank you, Ben!

A few of these should obviously be fixed in sources, but \uUsually, many of these issues are solved with a shell script (along with the build, or potentially in a fix + push to PR script). Do you think we should do that here, or make another mastering python script?

**Comment by benkiel** [comment]
We should just add it to the variable font build script, will do.

**Comment by arrowtype** [comment]
I've set the fsType and license string as of 714694c22e0750166c166812dc05ff85705fe066.

I'll short glyph names next. I don't think carets are really used by anything, but I'll add those as well.

**Comment by benkiel** [comment]
Ligature carets are used to place the cursor inside of a ligature glyph

**Comment by benkiel** [comment]
#314 adds in the `gasp`, `prep`, and `dsig` tables to the variable font build

**Comment by arrowtype** [comment]
Closing this issue – currently, ligature carets aren’t actually built into fonts generated from UFO sources. https://github.com/googlefonts/ufo2ft/issues/329

Additionally, programs tend to do a pretty good job of guessing where to place the caret inside ligatures. So, while this would be an issue for some fonts, it isn’t really a user problem for Recursive. If it later becomes possible to build carets into fonts and a user complains about this, I will open a fresh issue. This could be fixed relatively quickly with some scripting and a bit of manual work, but it seems pointless to spend time on unless the feature is supported and needed.

**Comment by davelab6** [comment]
Is the caret data in the sources? If yes, SGTM

**Comment by arrowtype** [comment]
Ligature carets are not currently in sources, and I’m happy to add them if they are useful. However, I am not aware of any issues caused by their exclusion. Are we adding them just for the sake of adding them?

**Comment by davelab6** [comment]
I often use Recursive as an 'exemplary' VF, and as I want to encourage
applications/OS to support correct caret positioning, having the GF library
support this data is helpful - that's why I had initially specified the FB
check as a FAIL. But I'm OK to deprioritize this effort, so, no need to add
unless you tend to ligatures again for some reason (like widths not being
consistent) and then I suggest to add the data in then.

**Comment by arrowtype** [comment]
Okay, that makes sense! Thanks for clarifying. If it may help other fonts that do need ligature carets to have proper support, then it will be good to add them here. I’ll work on this for the same release as fixing https://github.com/arrowtype/recursive/issues/431!

**Comment by arrowtype** [comment]
I’ve worked on a script that can (mostly) automate the addition of adding ligature carets across sources ([permalink to WIP script](https://github.com/arrowtype/recursive/commit/8507b5a01d2ad0ebeac160c4c9b32d1ae2fc7402)).

The last step here will be building all fonts, including and especially the code fonts, to verify that the added ligature carets don’t cause _new_ problems.

**Comment by arrowtype** [comment]
Closed by https://github.com/arrowtype/recursive/pull/434
